config/lib/seed.js

```diff
@@ -92,3 +65,85 @@ function reportError(err) {
   console.log(err);
   console.log();
 }
+
+function performSeedUser(user) {
+
+  var User = mongoose.model('User');
+
+  return new Promise(function(resolve, reject) {
+
+    var seedUser = {
+      username: 'user',
+      password: 'User_Password1!',
+      provider: 'local',
+      email: 'user@localhost.com',
+      firstName: 'User',
+      lastName: 'Local',
+      displayName: 'User Local',
+      roles: ['user']
+    };
+
+    var seedAdmin = {
+      username: 'admin',
+      password: 'Admin_Password1!',
+      provider: 'local',
+      email: 'admin@localhost.com',
+      firstName: 'Admin',
+      lastName: 'Local',
+      displayName: 'Admin Local',
+      roles: ['user', 'admin']
+    };
+
+    var user = null;
+
+    //If production only seed admin if it does not exist
+    if (process.env.NODE_ENV === 'production') {
+      //Add Local Admin
+
+      user = new User(seedAdmin);
+
+      User.generateRandomPassphrase()
```

Comment:
yeah I figured this is what you had in mind.
I'm not so sure I am in complete agreement about the readability, because that's just moving closing from one place to another. Just read again that block of code in saveUser and see how confusing that is - a function, that returns a function, that returns a promise. Goodluck debugging that :)